### PR TITLE
Favorite remove the route evaluation

### PR DESCRIPTION
**Pull request by mauriziovitale**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/app/components/toolbar/toggle-favorite/toggle-favorite.component.ts [hunk 1]
```diff
@@ -60,19 +60,13 @@ export class ToggleFavoriteComponent implements OnInit {
...
-    private store: Store<AppStore>,
-    private extensionService: ExtensionService,
-    private router: Router
-  ) {
+  constructor(private store: Store<AppStore>, private router: Router) {
     this.selection$ = this.store.select(getAppSelection);
   }
 
   ngOnInit() {
     if (this.data) {
```
Comment (DenysVuika) [inline src/app/components/toolbar/toggle-favorite/toggle-favorite.component.ts:68]: you don't need `this.data` I guess

Comment (mauriziovitale) [reply src/app/components/toolbar/toggle-favorite/toggle-favorite.component.ts:68]: refactored with JSON.Parse

**Comment by pionnegru** [comment]
I think there should be a unit test that will fail  with this changes

**Comment by mauriziovitale** [comment]
> I think there should be a unit test that will fail with this changes

Fixed with JSON.parse

**Pull request merged**